### DISCUSSION TITLE
Cannot upload image via API

### DISCUSSION THREAD

**Discussion by berkedel** [supabase/supabase#17837] [Questions] [answered]
I have a custom role `admin` in my project and configure the insert policy for my bucket below

```
((bucket_id = 'products'::text) AND (auth.role() = 'admin'::text))
```

Whenever I try to upload the image through the API, it always gives a 500 error. From the log, it seems the `supabase_storage_admin` user is trying to set the role for the admin but got denied.

```
Event message: permission denied to set role "admin"
Severity: ERROR
Timestamp: 2023-10-02T04:37:39.377Z
Postgres Username: supabase_storage_admin
Session ID: 651a47f2.14c02d
```

Anyone know what causing this error and how to fix this?

**Comment by GaryAustin1** [comment, marked as answer]
A new version of storage has broken custom roles. There is an open issue on it and Supabase has responded:  https://github.com/supabase/storage-api/issues/369

**Reply by berkedel** [reply]
Thank you for a quick answer 🙏
